felles/abac/src/main/java/no/nav/vedtak/sikkerhet/abac/PepImpl.java

```diff
@@ -56,17 +64,40 @@ public Tilgangsbeslutning vurderTilgang(BeskyttetRessursAttributter beskyttetRes
         if (PIP.equals(beskyttetRessursAttributter.getResourceType())) {
             return vurderTilgangTilPipTjeneste(beskyttetRessursAttributter, appRessurser);
         }
+        if (skalForetaLokalTilgangsbeslutning(beskyttetRessursAttributter)) {
+            return new Tilgangsbeslutning(harTilgang(beskyttetRessursAttributter) ? GODKJENT : AVSLÅTT_ANNEN_ÅRSAK, beskyttetRessursAttributter, appRessurser);
+        }
         return pdpKlient.forespørTilgang(beskyttetRessursAttributter, builder.abacDomene(), appRessurser);
     }
 
+    // AzureAD CC kommer med sub som ikke ikke en bruker med vanlige AD-grupper og roller
+    // Token kan utvides med roles og groups - men oppsettet er langt fra det som er kjent fra STS mv.
+    // Kan legge inn filter på claims/roles intern og/eller ekstern.
+    private boolean skalForetaLokalTilgangsbeslutning(BeskyttetRessursAttributter attributter) {
+        var identType = Optional.ofNullable(attributter.getToken().getSluttBruker()).map(SluttBruker::getIdentType).orElse(null);
+        var consumer = Optional.ofNullable(attributter.getToken().getSluttBruker()).map(SluttBruker::getName).orElse(null);
+        return OpenIDProvider.AZUREAD.equals(attributter.getToken().getOpenIDProvider())
+            && IdentType.Systemressurs.equals(identType) && consumer != null && preAuthorized != null;
+    }
+
+    private boolean harTilgang(BeskyttetRessursAttributter attributter) {
+        var consumer = Optional.ofNullable(attributter.getToken().getSluttBruker()).map(SluttBruker::getName).orElse(null);
+        if (consumer == null || !preAuthorized.contains(consumer)) {
+            return false;
+        }
+        if (consumer.startsWith(residentClusterNamespace) || builder.internAzureConsumer(consumer)) {
+            return true;
+        }
+        return AvailabilityType.ALL.equals(attributter.getAvailabilityType());
```

Comment:
Bør vi ikke sjekke om `preAuthorized.contains(consumer)` her også? Eller antar du at om de første to if() ikke treffer så er dette et ekstern kall så det er nok å sjekke på availability om APIet tillater alle??

Reply:
Den første gir deny hvis consumer ikke er preauth